cargo/cargo_build_script_runner/bin.rs

```diff
@@ -16,48 +16,77 @@
 // by rust_library/rust_binary.
 extern crate cargo_build_script_output_parser;
 
-use cargo_build_script_output_parser::BuildScriptOutput;
+use cargo_build_script_output_parser::{BuildScriptOutput, CompileAndLinkFlags};
 use std::env;
-use std::fs::{File, canonicalize, create_dir_all};
-use std::io::Write;
+use std::fs::{canonicalize, create_dir_all, write};
+use std::path::Path;
 use std::process::{exit, Command};
 
 fn main() {
+    let exec_root = env::current_dir().expect("Failed to get current directory");
+
     let mut args = env::args().skip(1);
     let manifest_dir_env = env::var("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR was not set");
-    let out_dir_env = env::var("OUT_DIR").expect("OUT_DIR was not set");
-    // For some reason RBE does not creat the output directory, force create it
-    create_dir_all(out_dir_env.clone()).expect(&format!("Failed to create OUT_DIR: {}", out_dir_env));
     let rustc_env = env::var("RUSTC").expect("RUSTC was not set");
     // Because of the Bazel's sandbox, bazel cannot provide full path, convert all relative path to correct path.
     let manifest_dir = canonicalize(&manifest_dir_env).expect(&format!("Failed to canonicalize '{}'", manifest_dir_env));
-    let out_dir = canonicalize(&out_dir_env).expect(&format!("Failed to canonicalize '{}'", out_dir_env));
     let rustc = canonicalize(&rustc_env).expect(&format!("Failed to canonicalize '{}'", rustc_env));
-    match (args.next(), args.next(), args.next(), args.next(), args.next()) {
-        (Some(progname), Some(crate_name), Some(envfile), Some(flagfile), Some(depenvfile)) => {
-            let output = BuildScriptOutput::from_command(
-                    Command::new(
-                        canonicalize(&progname).expect(&format!("Failed to canonicalize '{}'", progname)))
+
+    let cc = env::var_os("CC").map(|env_var| {
+        let cc_path = Path::new(&env_var);
+        if cc_path.is_relative() {
+            exec_root.join(cc_path).into_os_string()
+        } else {
+            env_var
+        }
+    });
+
+    match (args.next(), args.next(), args.next(), args.next(), args.next(), args.next(), args.next(), args.next()) {
+        (Some(progname), Some(crate_name), Some(out_dir_archive), Some(out_dir_path_file), Some(envfile), Some(flagfile), Some(linkflags), Some(depenvfile)) => {
+            let out_dir = Path::new(&out_dir_archive).parent().expect(&format!("out_dir_archive {:?} didn't have a parent", out_dir_archive)).join("out_dir");
+            create_dir_all(&out_dir).expect(&format!("Failed to create directory {:?}", out_dir));
+            write(&out_dir_path_file, out_dir.to_string_lossy().as_bytes()).expect(&format!("Failed to write {:?}", &out_dir_path_file));
+            let out_dir_abs = exec_root.join(&out_dir);
```

Comment:
Why not use `canonicalize`?

Reply:
See https://github.com/bazelbuild/rules_rust/pull/343#discussion_r441751274 - resolving symlinks isn't a great thing to implicitly do where what we really want is just an absolute path.